File: starlore-back/src/main/java/com/robin/blogback/entity/Category.java
package com.robin.blogback.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("categories")
public class Category {
    @TableId(type = IdType.AUTO)
    private Integer id;
    @TableField("user_id")
    private Integer userId;
    private String name;
    private String description;
    private String color;
    @TableField("article_count")
    private Integer articleCount;
    @TableField("createdAt")
    private LocalDateTime createdAt;
    @TableField("updatedAt")
    private LocalDateTime updatedAt;
}

File: starlore_app/lib/models/category.dart
/// 分类
class Category {
  final int id;
  final String name;
  final String? description;
  final String? color;
  final int articleCount;

  const Category({
    required this.id,
    required this.name,
    this.description,
    this.color,
    this.articleCount = 0,
  });

  factory Category.fromJson(Map<String, dynamic> json) {
    return Category(
      id: json['id'] ?? 0,
      name: json['name'] ?? '',
      description: json['description'],
      color: json['color'],
      articleCount: json['article_count'] ?? json['articleCount'] ?? 0,
    );
  }
}
